app/src/main/java/com/mapbox/maps/testapp/examples/RawExpressionActivity.kt

```diff
@@ -42,23 +43,19 @@ class RawExpressionActivity : AppCompatActivity() {
   }
 
   private fun addExpressionToStyle(style: Style) {
-    val expression = Value.fromJson(
-      """
-      [
-        "interpolate",
-         ["linear"],
-         ["zoom"],
-         0, "hsl(103, 80%, 70%)",
-         8.53, "hsl(360, 80%, 70%)",
-         22, "hsl(196, 80%, 70%)"
-       ]  
-      """.trimIndent()
+    style.getLayerAs<FillLayer>("water")?.fillColor(
+      Expression.fromRaw(
+        """
```

Comment:
To understand the context behind the Expression APIs better - were there thoughts to build type-safe DSL on top of this string-based fromRaw api?

Reply:
we have it already 😄 This activity just demonstrates how to use the raw expression as string.

Reply:
Thanks for the hint!